### PR TITLE
Updating GetBaseClass

### PR DESCRIPTION
**Pull request by Smit1603**

### CODE REVIEW

File: lib/Interpreter/CppInterOp.cpp [hunk 1]
```diff
@@ -615,11 +615,7 @@ namespace Cpp {
 
     auto type = (CXXRD->bases_begin() + ibase)->getType();
     if (auto RT = type->getAs<RecordType>()) {
```
Comment (vgvassilev) [inline lib/Interpreter/CppInterOp.cpp:617]:
```suggestion
    if (auto RT = type->getAs<RecordType>())
```

File: lib/Interpreter/CppInterOp.cpp [hunk 2]
```diff
@@ -615,11 +615,7 @@ namespace Cpp {
...
-      return (TCppScope_t) RT->getDecl();
-    } else if (auto TST = type->getAs<clang::TemplateSpecializationType>()) {
-      TemplateName TN = TST->getTemplateName();
-      if (auto *TD = dyn_cast_or_null<ClassTemplateDecl>(TN.getAsTemplateDecl()))
-        return TD->getTemplatedDecl();
+      return (TCppScope_t)RT->getDecl();
     }
```
Comment (vgvassilev) [inline lib/Interpreter/CppInterOp.cpp:619]:
```suggestion
```

Comment (Smit1603) [reply lib/Interpreter/CppInterOp.cpp]: Made all suggested changes

**Comment by vgvassilev** [comment]
The commit message should say what the change was done for instead of which function it was in. How about something like `Remove dead code since TemplateSpecializationType is a subclass for RecordType`.

**Pull request merged**